sql: audit production callers of `rowenc.DatumToEncDatum`

yuzefovich opened the pull request:
**sql: audit production callers of `rowenc.DatumToEncDatum`**

This commit audits all callers of `rowenc.DatumToEncDatum` in order to replace possibly unsafe calls to `DatumToEncDatumEx` which requires explicit error handling (the former panics when the datum's resolved type is not equivalent to the passed-in type). I decided to not remove the former method altogether since it seems to be reasonable to use it when we're construcing the datum directly in the call site (and also in tests).

Fixes: #154607.

**rowenc: rename DatumToEncDatum methods**

This commit does the following renaming:
- `DatumToEncDatumEx` -> `DatumToEncDatum`
- `DatumToEncDatum` -> `DatumToEncDatumUnsafe`.

The goal is to encourage usage of safer version which requires explicit error handling (as opposed to panicking). The main use case (apart from tests) for the unsafe version has been documented.

Release note: None

yuzefovich committed:
sql: audit production callers of `rowenc.DatumToEncDatum`

This commit audits all callers of `rowenc.DatumToEncDatum` in order to
replace possibly unsafe calls to `DatumToEncDatumEx` which requires
explicit error handling (the former panics when the datum's resolved
type is not equivalent to the passed-in type). I decided to not remove
the former method altogether since it seems to be reasonable to use it
when we're construcing the datum directly in the call site (and also in
tests).

Release note: None

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @mgartner reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154878#-:-Oaz12D4DYfEorfuC5h2:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball)

___
*[`pkg/sql/rowenc/encoded_datum.go` line 186 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/154878#-Oaz0PeNAHOJ012UeFZz:-Oaz0PeNAHOJ012UeF_-:bdhlmiq) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0aab48621843e0d01aabe9ad238cff2a094fbec/pkg/sql/rowenc/encoded_datum.go#L186)):*
> ```Go
> // NB: the method will panic if the datum's resolved type is not equivalent to
> // the passed-in type. Consider using safer DatumToEncDatumEx.
> func DatumToEncDatum(ctyp *types.T, d tree.Datum) EncDatum {
> ```

What if we named this `DatumToEncDatumUnsafe` to make folks think twice before using it and hopefully read the docs, and use the name `DatumToEncDatum` for the other version that returns an error?


<!-- Sent from Reviewable.io -->

yuzefovich committed:
rowenc: rename DatumToEncDatum methods

This commit does the following renaming:
- `DatumToEncDatumEx` -> `DatumToEncDatum`
- `DatumToEncDatum` -> `DatumToEncDatumUnsafe`.

The goal is to encourage usage of safer version which requires explicit
error handling (as opposed to panicking). The main use case (apart from
tests) for the unsafe version has been documented.

Release note: None

yuzefovich reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154878#-:-OazdLZS1mrXItiPYC3W:b-oduykh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @DrewKimball, @mgartner, and @msbutler)

___
*[`pkg/sql/rowenc/encoded_datum.go` line 186 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/154878#-Oaz0PeNAHOJ012UeFZz:-OazcgYU68gmEzVHg_G7:b-q1tve7) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0aab48621843e0d01aabe9ad238cff2a094fbec/pkg/sql/rowenc/encoded_datum.go#L186)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

What if we named this `DatumToEncDatumUnsafe` to make folks think twice before using it and hopefully read the docs, and use the name `DatumToEncDatum` for the other version that returns an error?
</blockquote></details>

Nice, I like it, done.


<!-- Sent from Reviewable.io -->

mgartner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @mgartner reviewed 60 of 60 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154878#-:-OazjSFO4zjUdE6dz1D1:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball and @msbutler)



<!-- Sent from Reviewable.io -->
